sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -96,37 +115,35 @@
      */
     private final File mLargePayloadDirectory;
 
-    /**
-     * Initializes variables.
-     */
-    public DatabasePersistence() {
-        this(DATABASE, TABLE, 1);
-    }
-
     /**
      * Initializes variables.
      *
-     * @param database The database name
-     * @param table    The table name
-     * @param version  The version of current schema.
+     * @param context application context.
      */
-    DatabasePersistence(String database, String table, @SuppressWarnings("SameParameterValue") int version) {
-        this(database, table, version, Persistence.DEFAULT_CAPACITY);
+    public DatabasePersistence(Context context) {
+        this(context, 2, SCHEMA, Persistence.DEFAULT_CAPACITY);
     }
 
     /**
      * Initializes variables.
      *
-     * @param database   The database name
-     * @param table      The table name
+     * @param context    application context.
      * @param version    The version of current schema.
+     * @param schema     schema.
      * @param maxRecords The maximum number of records allowed in the table.
      */
-    DatabasePersistence(String database, String table, int version, int maxRecords) {
+    DatabasePersistence(Context context, int version, ContentValues schema, int maxRecords) {
+        mContext = context;
         mPendingDbIdentifiersGroups = new HashMap<>();
         mPendingDbIdentifiers = new HashSet<>();
-        mDatabaseStorage = DatabaseStorage.getDatabaseStorage(database, table, version, SCHEMA, maxRecords,
-                new DatabaseStorage.DatabaseErrorListener() {
+        mDatabaseStorage = DatabaseStorage.getDatabaseStorage(DATABASE, TABLE, version, schema, maxRecords,
+                new DatabaseManager.Listener() {
+
+                    @Override
+                    public boolean onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+                        db.execSQL("ALTER TABLE " + TABLE + " ADD COLUMN `" + COLUMN_TARGET_TOKEN + "` TEXT");
```

Comment:
Add versions check.

Reply:
This is not called on downgrade and for this version we don't need checks yet.
We can introduce version check the next time we change schema, would be dead code to cover in unit tests otherwise, I will add a comment instead.